### PR TITLE
pcode2code: initial commit; close #2569

### LINKED ISSUE #2569: [Tool request] pcode2code
https://github.com/Big5-sec/pcode2code

### PR DESCRIPTION
**Pull request by demogorgonz**
I have used dos2unix to convert CR characters within pcode2code, any other preferred way to achieve this ?

### CODE REVIEW

File: packages/pcode2code/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+package() {
+  cd $pkgname
+  install -Dm 755 "$pkgname/$pkgname.py" "$pkgdir/usr/bin/$pkgname"
+  install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname/" README.md
+  install -Dm 644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
+  chmod +x "$pkgdir/usr/bin/$pkgname"
+  dos2unix "$pkgdir/usr/bin/$pkgname"
+}
```
Comment (Edu4rdSHL) [inline packages/pcode2code/PKGBUILD]: Add final new line to fix travis.

File: packages/pcode2code/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# See COPYING for license details.
+
+pkgname=pcode2code
+pkgver=4.4de2193
+pkgrel=1
+groups=('blackarch' 'blackarch-decompiler')
+pkgdesc='VBA p-code decompiler'
+url='https://github.com/Big5-sec/pcode2code'
+arch=('any')
+license=('GNU3')
```
Comment (Edu4rdSHL) [inline packages/pcode2code/PKGBUILD]: Add a valid license.

File: packages/pcode2code/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+sha512sums=('SKIP')
+
+pkgver() {
+  cd $pkgname
+
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+}
+
+prepare() {
+  cd $pkgname
```
Comment (Edu4rdSHL) [inline packages/pcode2code/PKGBUILD]: Add newline.

File: packages/pcode2code/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  sed -i 's|python|python3|g' "$pkgname/$pkgname.py"
+}
+
+package() {
+  cd $pkgname
+  install -Dm 755 "$pkgname/$pkgname.py" "$pkgdir/usr/bin/$pkgname"
+  install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname/" README.md
+  install -Dm 644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
+  chmod +x "$pkgdir/usr/bin/$pkgname"
```
Comment (Edu4rdSHL) [inline packages/pcode2code/PKGBUILD]: Remove it line.

File: packages/pcode2code/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  dos2unix "$pkgdir/usr/bin/$pkgname"
```
Comment (Edu4rdSHL) [inline packages/pcode2code/PKGBUILD]: Remove it line.

File: packages/pcode2code/PKGBUILD (hunk 4 above)
Comment (demogorgonz) [reply packages/pcode2code/PKGBUILD]: Do you mean sed part?

Comment (Edu4rdSHL) [reply packages/pcode2code/PKGBUILD]: The chmod thing.

**Comment by ikstream** [comment]
Please have a look at blackarch#2021 for naming convention for future PRs

**Pull request merged**